packages/gatsby-link/index.d.ts

```diff
@@ -1,11 +1,21 @@
-import * as React from "react";
+import React from "react"
+import { LinkProps } from "reach__router"
 
-export const push: (to: LocationDescriptor) => void;
-export const replace: (to: LocationDescriptor) => void;
+export const push: (to: string) => void
+export const replace: (to: string) => void
```

Comment:
These two functions are actually deprecated as well in v2 in favour of `navigate`. You can add the following to the declaration:

```ts
import { NavigateFn } from "@reach/router";

export const navigate: NavigateFn;
```